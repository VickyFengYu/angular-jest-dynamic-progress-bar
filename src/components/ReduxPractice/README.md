# ReduxPractice

```js
<ReduxPractice>This is my component</ReduxPractice>
```
