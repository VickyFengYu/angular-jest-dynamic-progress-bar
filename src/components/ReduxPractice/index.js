import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import App from './components/App'
import rootReducer from './reducers'


const store = createStore(rootReducer)

class ReduxPractice extends Component {
  render() {

    // <div className={this.props.className}>{this.props.children}</div>;
    return <Provider store={store}>
      <App />
    </Provider>;

  }
}

ReduxPractice.defaultProps = {
  className: 'ReduxPractice'
};

ReduxPractice.propTypes = {
  /** Class name */
  className: PropTypes.node,
  /** Children inside component */
  children: PropTypes.node
};

export default ReduxPractice;

// import React from 'react'
// import { render } from 'react-dom'
// import { createStore } from 'redux'
// import { Provider } from 'react-redux'
// import App from './components/App'
// import rootReducer from './reducers'

// const store = createStore(rootReducer)

// render(
//   <Provider store={store}>
//     <App />
//   </Provider>,
//   document.getElementById('root')
// )


