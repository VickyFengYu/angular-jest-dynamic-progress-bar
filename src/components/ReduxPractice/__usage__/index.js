import React from 'react';
import { Heading } from '@lux/components';
import NewReduxPractice from '..';

const TestNewReduxPracticeStory = () => (
    <NewReduxPractice
    >
        <Heading level={1}>{"Vicky's New Redux Practice"}</Heading>
    </NewReduxPractice>
);

export default TestNewReduxPracticeStory;
export const name = "NewReduxPractice";