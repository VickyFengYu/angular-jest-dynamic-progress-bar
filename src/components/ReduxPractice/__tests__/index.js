import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ReduxPractice from '../index';

import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() })

describe('ReduxPractice Component', () => {
  it('should shallow render', () => {
    const wrapper = shallow(<ReduxPractice />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
