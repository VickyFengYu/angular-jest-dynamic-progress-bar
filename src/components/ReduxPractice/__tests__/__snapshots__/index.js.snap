// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReduxPractice Component should shallow render 1`] = `
<div
  className="ReduxPractice"
/>
`;
